### PR TITLE
Print out RUNNING, PASSED, WAIVED, FAILED for each sub-test in sanity

### PR DESCRIPTION
**Pull request by pakmarkthub**
Problems:
- _sanity_ does not have clear labels to indicate the boundary of each sub test.
- This PR is part of the RFE discussed in nvbug 200565850.

This PR:
- prints RUNNING at the start of each sub test.
- prints either PASSED, WAIVED, or FAILED at the end of each sub test.
- replaces ck_assert* with ASSERT*.
- creates common.cpp as a container for common settings and functions.

### CODE REVIEW

File: tests/common.hpp [hunk 1]
```diff
@@ -80,6 +84,52 @@ namespace gdrcopy {
                 return CUDA_ERROR_INVALID_VALUE;
         }
 
+        bool print_dbg_msg = false;
```
Comment (drossetti) [inline tests/common.hpp]: you need static here

File: tests/common.hpp [hunk 2]
```diff
@@ -80,6 +84,52 @@ namespace gdrcopy {
...
+            if (print_dbg_msg) {
+                va_list ap;
+                va_start(ap, fmt);
+                vfprintf(stderr, fmt, ap);
+            }
+        }
+
+        #define EXIT_WAIVED 2
+
+        const char *testname = "";
```
Comment (drossetti) [inline tests/common.hpp]: same as above

**Comment by pakmarkthub** [comment]
Fixed the PR. @drossetti Please re-review.

File: tests/sanity.cpp [hunk 3]
```diff
@@ -56,18 +55,6 @@ using namespace gdrcopy::test;
 #define FENCE() asm volatile("":::"memory")
```
[the remark this answers is not in the document]
Comment (drossetti) [reply tests/sanity.cpp]: let's call it barrier

**Comment by pakmarkthub** [comment]
Thank you Davide.

Addressed the latest comment. Merging the PR.

**Pull request merged**